packages/interactive_media_ads/lib/src/ios/ios_ad_display_container.dart

```diff
@@ -0,0 +1,86 @@
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart';
+
+import '../platform_interface/platform_interface.dart';
+import 'interactive_media_ads.g.dart';
+import 'interactive_media_ads_proxy.dart';
+
+/// Implementation of [PlatformAdDisplayContainerCreationParams] for iOS.
+final class IOSAdDisplayContainerCreationParams
+    extends PlatformAdDisplayContainerCreationParams {
+  /// Constructs a [IOSAdDisplayContainerCreationParams].
+  const IOSAdDisplayContainerCreationParams({
+    super.key,
+    required super.onContainerAdded,
+    @visibleForTesting InteractiveMediaAdsProxy? imaProxy,
+  })  : _imaProxy = imaProxy ?? const InteractiveMediaAdsProxy(),
+        super();
+
+  /// Creates a [IOSAdDisplayContainerCreationParams] from an instance of
+  /// [PlatformAdDisplayContainerCreationParams].
+  factory IOSAdDisplayContainerCreationParams.fromPlatformAdDisplayContainerCreationParams(
+    PlatformAdDisplayContainerCreationParams params, {
+    @visibleForTesting InteractiveMediaAdsProxy? imaProxy,
+  }) {
+    return IOSAdDisplayContainerCreationParams(
+      key: params.key,
+      onContainerAdded: params.onContainerAdded,
+      imaProxy: imaProxy,
+    );
+  }
+
+  final InteractiveMediaAdsProxy _imaProxy;
+}
+
+/// Implementation of [PlatformAdDisplayContainer] for iOS.
+base class IOSAdDisplayContainer extends PlatformAdDisplayContainer {
+  /// Constructs an [IOSAdDisplayContainer].
+  IOSAdDisplayContainer(super.params) : super.implementation() {
+    _controller = _iosParams._imaProxy.newUIViewController();
+  }
+
+  // The `UIViewController` used to create the native `IMAAdDisplayContainer`.
+  late final UIViewController _controller;
+
+  /// The native iOS IMAAdDisplayContainer.
+  ///
+  /// Created with the `UIView` that handles playing an ad.
+  @internal
+  late final IMAAdDisplayContainer? adDisplayContainer;
+
+  late final IOSAdDisplayContainerCreationParams _iosParams =
+      params is IOSAdDisplayContainerCreationParams
+          ? params as IOSAdDisplayContainerCreationParams
+          : IOSAdDisplayContainerCreationParams
+              .fromPlatformAdDisplayContainerCreationParams(params);
+
+  @override
+  Widget build(BuildContext context) {
+    return UiKitView(
+      key: _iosParams.key,
+      viewType: 'interactive_media_ads.packages.flutter.dev/view',
+      onPlatformViewCreated: (_) async {
+        adDisplayContainer = _iosParams._imaProxy.newIMAAdDisplayContainer(
+          adContainer: _controller.view,
+          adContainerViewController: _controller,
+        );
+        // A delay is added since this callback only indicates when the View is
+        // created, but not when it has been added to the native View hierarchy.
+        // See https://github.com/flutter/flutter/issues/150802
+        await Future<void>.delayed(const Duration(seconds: 1));
```

Comment:
Do we think this is going to always be enough time?

Reply:
I changed it to wait for `UIViewController.viewDidAppear`. It turns out UIKit doesn't really support KVO and this was the correct way to do it.